crates/storage/db/src/implementation/mdbx/mod.rs

```diff
@@ -66,14 +66,15 @@ impl<E: EnvironmentKind> Env<E> {
         path: &Path,
         kind: EnvKind,
         log_level: Option<LogLevel>,
+        max_tables: usize,
     ) -> Result<Env<E>, DatabaseError> {
         let mode = match kind {
             EnvKind::RO => Mode::ReadOnly,
             EnvKind::RW => Mode::ReadWrite { sync_mode: SyncMode::Durable },
         };
 
         let mut inner_env = Environment::new();
-        inner_env.set_max_dbs(Tables::ALL.len());
+        inner_env.set_max_dbs(max_tables);
```

Comment:
Ah! Nice! @shekhirin @onbjerg @rkrasiuk we've talked about this before, lifts the limit from having a comp-time defined set of stages